### DISCUSSION TITLE
Cache Options not work with node api

### DISCUSSION THREAD

**Discussion by childrentime** [webpack/webpack#17342] [Q&A] [answered]
# Bug report

**What is the current behavior?**

cache option will not work with webpack ndoejs api, like below

```js
const webpack  = require('webpack');
const config = require('./webpack.config');

webpack(config).run((stat) => console.log(stat))
```

will not generate .cache in node_modules

Howerver, Use `npm run dev`+`webpack --config.js` works well

**If the current behavior is a bug, please provide the steps to reproduce.**

checkout this https://github.com/childrentime/webpack-cache-test

**What is the expected behavior?**

generate cache file as expect

**Other relevant information:**

version: latest

**Comment by childrentime** [comment]
btw, watch function will genarate cache

**Reply by alexander-akait** [reply, marked as answer]
You need to run `close`, otherwise webpack can't store cache, here and example in our docs https://webpack.js.org/api/node/#run

**Comment by childrentime** [comment]
sorry forgot this in document🥲! Thank you🥵
